addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/NeatoRobot.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neato.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.TimeZone;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Hex;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.neato.handler.NeatoHandler;
+import org.openhab.binding.neato.internal.classes.ErrorMessage;
+import org.openhab.binding.neato.internal.classes.NeatoGeneralInfo;
+import org.openhab.binding.neato.internal.classes.NeatoRobotInfo;
+import org.openhab.binding.neato.internal.classes.NeatoState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link NeatoBinding} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+
+public class NeatoRobot {
+
+    private static final Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
```

Comment:
Logger should NOT be static according to the guidelines. [[source](http://docs.openhab.org/developers/development/guidelines.html#e-logging)] and the wrong class is passed to the logger.